Fix configuration and Travis errors

OskarPersson opened the pull request:
Fixes regressions from #150 and #151

peppelinux commented:
Still waiting here
https://github.com/knaperek/djangosaml2/pull/169/files

OskarPersson commented:
Oops, completely missed that one. Still relevant for #150 though (right?)

OskarPersson commented:
Also @knaperek, I'm not sure why these PRs (including #166) was merged when Travis failed? I think you should configure this repo to

1) report travis output on the PR pages and
2) disallow merging of PRs where travis has failed

OskarPersson commented:
The [Travis build](https://travis-ci.org/knaperek/djangosaml2/builds/638516394) still fails on Django 2.2+, though I can't reproduce the error locally.

peppelinux commented:
> Oops, completely missed that one. Still relevant for #150 though (right?)

Not Shure, selected_idps Will be Always false in that position, so the if Is useless I think.
Thank you for the feedback Oscar

peppelinux commented:
> The [Travis build](https://travis-ci.org/knaperek/djangosaml2/builds/638516394) still fails on Django 2.2+, though I can't reproduce the error locally.

auth_group schema does not work there.
We should also remove support for py35.
I can do a pr on this but I'll take two weeks from now

OskarPersson commented:
I've now updated the Travis dist to get rid of the schema errors and simultanesly dropped support for Python 3.4 (EOL 2019-03-18): I've kept Python 3.5 which has its EOL 2020-09-13:

https://devguide.python.org/devcycle/#end-of-life-branches
https://devguide.python.org/#status-of-python-branches

Travis build: https://travis-ci.org/knaperek/djangosaml2/builds/638821120

@knaperek This PR now contains a mixed bag of fixes, are you okay with merging it as-is or should I split it up into multiple PRs first?

peppelinux commented:
Nice, of my point of view Means something here:

forceAuthn and allowcreate patch should be refactored in the way I explained

The same for selected_idp

Old versions of Django could be dropped as well

OskarPersson commented:
Yeah you definitely know more of that stuff than me, I will remove those changes from this branch

peppelinux commented:
this also should be refactored: https://github.com/knaperek/djangosaml2/pull/167/files

OskarPersson commented:
Regarding the change of selected_idp, if I remove the if-statement I get the following error from the tests:

```
FAIL: test_login_several_idps (djangosaml2.tests.SAML2Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/oskar/git/djangosaml2/.tox/py37-django22/lib/python3.7/site-packages/djangosaml2/tests/__init__.py", line 197, in test_login_several_idps
    self.assertEqual(url.hostname, 'idp2.example.com')
AssertionError: 'idp3.example.com' != 'idp2.example.com'
- idp3.example.com
?    ^
+ idp2.example.com
? 
```

peppelinux commented:
> Regarding the change of selected_idp, if I remove the if-statement I get the following error from the tests:
> 
> ```
> FAIL: test_login_several_idps (djangosaml2.tests.SAML2Tests)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/oskar/git/djangosaml2/.tox/py37-django22/lib/python3.7/site-packages/djangosaml2/tests/__init__.py", line 197, in test_login_several_idps
>     self.assertEqual(url.hostname, 'idp2.example.com')
> AssertionError: 'idp3.example.com' != 'idp2.example.com'
> - idp3.example.com
> ?    ^
> + idp2.example.com
> ? 
> ```

Ok @OskarPersson That's important, my excuses here.

OskarPersson commented:
@knaperek Anything missing for merge?

knaperek commented:
All good, tack så mycket!
